packages/grid/src/vaadin-grid-a11y-mixin.js

```diff
@@ -34,21 +34,27 @@ export const A11yMixin = (superClass) =>
 
     /** @private */
     _a11yGetFooterRowCount(_columnTree) {
-      return _columnTree.filter((level) => level.some((col) => col.headerRenderer)).length;
+      return _columnTree.filter((level) => level.some((col) => col.footerRenderer)).length;
     }
 
     /** @private */
-    _a11yUpdateGridSize(size, _columnTree) {
+    _a11yUpdateGridSize(size, _columnTree, __emptyState) {
       if (size === undefined || _columnTree === undefined) {
         return;
       }
 
+      const headerRowsCount = this._a11yGetHeaderRowCount(_columnTree);
+      const footerRowsCount = this._a11yGetFooterRowCount(_columnTree);
+      const bodyRowsCount = __emptyState ? 1 : size;
+      const rowsCount = bodyRowsCount + headerRowsCount + footerRowsCount;
+
+      this.$.table.setAttribute('aria-rowcount', rowsCount);
+
       const bodyColumns = _columnTree[_columnTree.length - 1];
-      this.$.table.setAttribute(
-        'aria-rowcount',
-        size + this._a11yGetHeaderRowCount(_columnTree) + this._a11yGetFooterRowCount(_columnTree),
-      );
-      this.$.table.setAttribute('aria-colcount', (bodyColumns && bodyColumns.length) || 0);
+      // If no header and footer rows while the empty state is active, count as one column
+      // Otherwise, use the number of body columns, if present
+      const columnsCount = __emptyState && rowsCount === 1 ? 1 : (rowsCount && bodyColumns && bodyColumns.length) || 0;
```

Comment:
Shouldn't the number of columns always be 1 in the empty state regardless of the row count?

```suggestion
      const columnsCount = __emptyState ? 1 : (rowsCount && bodyColumns && bodyColumns.length) || 0;
```

Reply:
It looks like the change didn't go through, as it's still showing `rowsCount === 1`

Reply:
It might have been some merge conflicts I had during the edition.